sdk/core/src/pipeline.rs

```diff
@@ -51,17 +52,33 @@ impl Pipeline {
 
         pipeline.extend_from_slice(&per_call_policies);
         pipeline.extend_from_slice(&options.per_call_policies);
-        pipeline.push(Arc::new(TelemetryPolicy::new(
-            crate_name,
-            crate_version,
-            &options.telemetry,
-        )));
-        pipeline.push(retry);
+
+        let telemetry_policy = TelemetryPolicy::new(crate_name, crate_version, &options.telemetry);
+        pipeline.push(Arc::new(telemetry_policy));
+
+        let retry_policy = options.retry.to_policy();
+        pipeline.push(retry_policy);
+
         pipeline.extend_from_slice(&per_retry_policies);
         pipeline.extend_from_slice(&options.per_retry_policies);
-        pipeline.push(transport_policy);
 
-        Self { pipeline }
+        // TODO: Add transport policy for WASM once https://github.com/Azure/azure-sdk-for-rust/issues/293 is resolved.
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+            let transport_policy = TransportPolicy::new(&options.transport);
+            pipeline.push(Arc::new(transport_policy));
+        }
+
+        Self {
+            http_client: options.transport.http_client.clone(),
+            pipeline,
+        }
+    }
+
+    /// Gets the `HttpClient` used by the pipeline.
+    pub fn http_client(&self) -> &dyn HttpClient {
+        // TODO: Request methods should be defined directly on the pipeline instead of exposing the HttpClient.
```

Comment:
Can you create an issue for this?

Reply:
Ah I see that there's this: https://github.com/Azure/azure-sdk-for-rust/issues/294. Might be worth in the future having all todos link to an issue